pkg/component/kubernetes/apiserver/deployment.go

```diff
@@ -536,9 +553,9 @@ func (k *kubeAPIServer) handleVPNSettings(
 	}
 
 	if k.values.VPN.HighAvailabilityEnabled {
-		k.handleVPNSettingsHA(deployment, serviceAccount, secretCAVPN, secretHAVPNSeedClient, secretHAVPNSeedClientSeedTLSAuth)
+		return k.handleVPNSettingsHA(deployment, serviceAccount, secretCAVPN, secretHTTPProxyClient, secretHTTPProxy, secretHAVPNSeedClient, secretHAVPNSeedClientSeedTLSAuth, configMapEgressSelector, configMapEnvoyConfig)
 	} else {
-		k.handleVPNSettingsNonHA(deployment, secretCAVPN, secretHTTPProxy, configMapEgressSelector)
+		k.handleVPNSettingsNonHA(deployment, secretCAVPN, secretHTTPProxyClient, configMapEgressSelector)
 	}
 
 	return nil
```

Comment:
From my point of view, that adding the `return` in the `if` block changes the logic in a surprising way. Please consider removing the `else` block or make it symmetric.

```suggestion
	if k.values.VPN.HighAvailabilityEnabled {
		return k.handleVPNSettingsHA(deployment, serviceAccount, secretCAVPN, secretHTTPProxyClient, secretHTTPProxy, secretHAVPNSeedClient, secretHAVPNSeedClientSeedTLSAuth, configMapEgressSelector, configMapEnvoyConfig)
	}

	k.handleVPNSettingsNonHA(deployment, secretCAVPN, secretHTTPProxyClient, configMapEgressSelector)

	return nil
```

Reply:
I just noticed we actually don't throw any error in the ha version as well. I will just drop it.